src/lib/useTableResize.ts

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import type {Column, HandleResize} from './types';
+
+export type TableColumnsWidthSetup = Record<string, number>;
+
+export function updateColumnsWidth<T>(
+    columns: Column<T>[],
+    columnsWidthSetup: TableColumnsWidthSetup,
+    defaultResizeable = false,
+): Column<T>[] {
+    return columns.map((column) => {
+        let sub: Column<T>[] | undefined;
+
+        if (column.sub) {
+            sub = updateColumnsWidth(column.sub, columnsWidthSetup, defaultResizeable);
+        }
+
+        const newWidth = columnsWidthSetup[column.name] ?? column.width;
+
+        return {...column, width: newWidth, sub};
+    });
+}
+
+export function useTableResize(localStorageKey: string): [TableColumnsWidthSetup, HandleResize] {
```

Comment:
Current implementation promotes to use `localStorage` of browser and it doesn't looks like a good idea cause it has 5MB limit for stored data. And some services with many tables may overflow the limit in some time. It is better to require some abstract methods in parameters of the hook:
```ts
function useTableResize(
    params: {
        saveSizes: (data: TableColumnsWidthSetup) => void;
        getSizes: () => TableColumnsWidthSetup;
    }
): [TableColumnsWidthSetup, HandleResize] {
    ...
}```

Reply:
I assumed this hook as basic implementation with the possibility to write your own hook if needed (that is why it is a separate hook, but not some logic inside `DataTable`). But your solution event better, I'll use it